### PR TITLE
Created Dummy plugin with process method

### PR DESCRIPTION
**Pull request by laurisigaunis**
Created Dummy plugin to make sure plugin system is working.
Dummy plugin contains process method and into it creates Drupal logger notices for each item.

Trello - https://trello.com/c/csBqEozG/9-dummy-sapi-plugin

### CODE REVIEW

File: src/Plugin/StatisticsPluginInterface.php [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
 
 use Drupal\Component\Plugin\PluginInspectionInterface;
+use Drupal\sapi;
 
 /**
  * Defines an interface for Statistics plugin plugins.
  */
 interface StatisticsPluginInterface extends PluginInspectionInterface {
 
-  // Add get/set methods for your plugin type here.
```
Comment (james-nesbitt) [inline src/Plugin/StatisticsPluginInterface.php]: Full PHPDOC documentation required in the interface

File: src/Plugin/StatisticsPlugin/Dummy.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+use Drupal\sapi\Plugin\StatisticsPluginInterface;
+
+/**
+ * @StatisticsPlugin(
+ *  id = "dummy",
+ *  label = "Dummy label"
+ * )
+ */
+class Dummmy implements StatisticsPluginInterface {
+
```
Comment (james-nesbitt) [inline src/Plugin/StatisticsPlugin/Dummy.php]: @InheritDoc comment required here

**Comment by james-nesbitt** [comment]
your code is good, but you need more comments (PHPDOC stuff)

interface method needs to have a full PHPDOC definition:

```
/**
 * Short description of what I do
 *
 * @param {type} $argument
 *      I am a {type} that is magical
 * @returns {type}
 *     I return X
 * @throws {type}
 *     I may throw an exception if I am in a bad mood
 */
```

The actual plugin can just use 

```
  /**
   * {@inheritdoc}
   */
```

File: src/Plugin/StatisticsPlugin/Dummy.php [hunk 3]
```diff
@@ -12,4 +12,8 @@
  */
 class Dummmy implements StatisticsPluginInterface {
```
Comment (maijs) [inline src/Plugin/StatisticsPlugin/Dummy.php]: `Dummy` doesn't extend `StatisticsPluginBase` class (which it should).

File: src/Plugin/StatisticsPlugin/Dummy.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+use Drupal\sapi\Plugin\StatisticsPluginInterface;
+
+/**
+ * @StatisticsPlugin(
+ *  id = "dummy",
+ *  label = "Dummy label"
+ * )
+ */
+class Dummmy implements StatisticsPluginInterface {
```
Comment (maijs) [inline src/Plugin/StatisticsPlugin/Dummy.php]: `Dummy` doesn't extend `StatisticsPluginBase` class (which it should).

Comment (james-nesbitt) [inline src/Plugin/StatisticsPlugin/Dummy.php]: it's actually optional to use the base class.  it is only required that it use the interface

File: src/Plugin/StatisticsPluginInterface.php [hunk 5]
```diff
@@ -3,12 +3,16 @@
 namespace Drupal\sapi\Plugin;
 
 use Drupal\Component\Plugin\PluginInspectionInterface;
+use Drupal\sapi;
```
Comment (maijs) [inline src/Plugin/StatisticsPluginInterface.php]: `Drupal\sapi` is not used in the interface.

File: src/Plugin/StatisticsPluginInterface.php [hunk 6]
```diff
@@ -3,12 +3,16 @@
...
 /**
  * Defines an interface for Statistics plugin plugins.
  */
 interface StatisticsPluginInterface extends PluginInspectionInterface {
 
-  // Add get/set methods for your plugin type here.
+  /**
+   * process() method which analize the data received from dispatcher service
+   */
+  public function process(StatisticsItemIterface $item);
```
Comment (maijs) [inline src/Plugin/StatisticsPluginInterface.php:20]: `StatisticsItemIterface` is not imported using `use` statement, therefore the script will fail. I'd suggest adding `use Drupal\sapi\StatisticsItemInterface;`

File: src/Plugin/StatisticsPlugin/Dummy.php (hunk 4 above)
Comment (maijs) [inline src/Plugin/StatisticsPlugin/Dummy.php]: `StatisticsPluginInterface` implements `PluginInspectionInterface` and to comply with that the plugin needs to have `getPluginId()` and other methods which are provided by `PluginBase` class which is a parent of `StatisticsPluginBase`. So, unless the Plugin really wants to implement all those declared classes, it should extend `StatisticsPluginBase` or at least `PluginBase`.

**Comment by james-nesbitt** [comment]
it's good for me.

**Pull request merged**